### PR TITLE
Deprecating methods not used anymore on mule 4 branch

### PR DESCRIPTION
**Pull request by pablokraan**

### CODE REVIEW

File: core/src/main/java/org/mule/api/MuleEvent.java [hunk 1]
```diff
@@ -322,6 +322,10 @@
 
     void clearFlowVariables();
 
+    /**
+     * @deprecated use {@code getSession()} to manipulate session state
+     */
+    @Deprecated
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/api/MuleEvent.java]: shouldn't we wait to get sign off on the message changes spec before doing this?

**Comment by marianogonzalez** [comment]
Sounds to me that we shouldn't do this until the new message structure is merged in 4.x

File: core/src/main/java/org/mule/api/MuleEvent.java (hunk 1 above)
Comment (pablokraan) [inline core/src/main/java/org/mule/api/MuleEvent.java]: These changes are not related to the changes we discussed about changing the message/event. This is just cleaning things that are not used anymore or must not be used
